src/main/java/org/spongepowered/common/registry/type/StatisticFormatRegistryModule.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.registry.type;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Maps;
+import net.minecraft.stats.StatList;
+import org.spongepowered.api.registry.AdditionalCatalogRegistryModule;
+import org.spongepowered.api.registry.util.AdditionalRegistration;
+import org.spongepowered.api.registry.util.RegisterCatalog;
+import org.spongepowered.api.statistic.Statistic;
+import org.spongepowered.api.statistic.StatisticFormat;
+import org.spongepowered.api.statistic.StatisticFormats;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+
+public final class StatisticFormatRegistryModule implements AdditionalCatalogRegistryModule<StatisticFormat> {
+
+    @RegisterCatalog(StatisticFormats.class)
+    private final Map<String, StatisticFormat> formatMappings = Maps.newHashMap();
+
+    @Override
+    public Optional<StatisticFormat> getById(String id) {
+        return Optional.ofNullable(this.formatMappings.get(checkNotNull(id).toLowerCase()));
+    }
+
+    @Override
+    public Collection<StatisticFormat> getAll() {
+        return ImmutableList.copyOf(this.formatMappings.values());
+    }
+
+    @Override
+    public void registerDefaults() {
+        StatList.allStats.stream()
+                .map(statistic -> (Statistic) statistic)
```

Comment:
You can combine the cast with the call below, like `.map(stat -> ((Statistic) stat).getStatisticFormat()...`